### PR TITLE
[16.0] [MIG] iot_oca: Migration to 16.0

### PR DESCRIPTION
**Pull request by mostafabarmshory**
I try to migrate IoT module from 15 to 16. Here is the guide line

https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0

### CODE REVIEW

**Comment by mostafabarmshory** [comment]
Dear @etobella and @pedrobaeza :

I followed the guideline and moved a single (core addon). Others depend on this and I have to wait for it. Would you check and accept the pull? please.

**Comment by mostafabarmshory** [comment]
Dear @etobella ,
Would you check the PR and provide feedback to me, please? I`ve followed instructions step by step. Other add-ons depend on this one and I need it to continue.

File: iot_oca/tests/test_iot.py [hunk 1]
```diff
@@ -1,6 +1,13 @@
 # Copyright (C) 2018 Creu Blanca
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
-from mock import patch
+
+try:
+    from mock import patch
+except ImportError:
+    import os
+
+    os.system("pip install mock")
```
Comment (etobella) [inline iot_oca/tests/test_iot.py]: Don't do that

Keep as 

`from mock import patch`

If you don't have this installed, the only problem will be a failure in test, and that is not important.

**Review by etobella — CHANGES_REQUESTED** [review]

File: iot_oca/tests/test_iot.py (hunk 1 above)
Comment (mostafabarmshory) [reply iot_oca/tests/test_iot.py]: Ok, bro.
Tests failed in my development environment. I guess they fail on GitHub too. 

np, I fix this right now.

**Comment by mostafabarmshory** [comment]
Dear @etobella 
as you see all test fails. We need the mock module just in the development environment. I can't find a way to add the dependency just for tests. 

Could you help me to solve the issue, please?

**Comment by etobella** [comment]
pre-commit fails due to some linting errors. intstall pre-commit on your environment.

If you need a testing library add it to test-requirements.txt file on the main folder. Something like this:

https://github.com/OCA/purchase-workflow/blob/c186f0808b932416bb5e1cab563cfd87c2ee4d3b/test-requirements.txt#L1

**Comment by mostafabarmshory** [comment]
@etobella 
finally, all checks passed.

**Comment by etobella** [comment]
Can you squash all your commits?

**Comment by mostafabarmshory** [comment]
Closed due to https://github.com/OCA/iot/pull/85

**Pull request closed without merging**